vault-plugins/flant_iam_auth/backend/path_auth_method.go

```diff
@@ -340,6 +340,9 @@ func (b *flantIamAuthBackend) pathAuthMethodCreateUpdate(ctx context.Context, re
 		if methodType == model.MethodTypeOIDC && !(authType == model.AuthSourceOIDCFlow || authType == model.AuthSourceOIDCDiscovery) {
 			return logical.ErrorResponse(fmt.Sprintf("incorrect source '%s': need OIDC based source", sourceName)), nil
 		}
+		if methodType == model.MethodTypeAccessToken && !(authType == model.AuthSourceOIDCDiscovery) {
+			return logical.ErrorResponse(fmt.Sprintf("incorrect source '%s': need oidc_discovery_url based source", sourceName)), nil
+		}
```

Comment:
because github cannot comment unchanged line, write here.

You should add new type into if's with verifying source name
https://github.com/flant/negentropy/pull/267/files#diff-21dfbd6faa3b78a64f6a76a97d8be637235ccba4ab0f8126e77e77b1a77d9db0R606

Reply:
github is broke. this function in main https://github.com/flant/negentropy/blob/7a7e1e0ab5b117b6ce1fe80078bee961eadf5bb6/vault-plugins/flant_iam_auth/backend/path_auth_method.go#L596

Reply:
done